Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/api/ui/TestForAPIFragment.kt

```diff
@@ -258,34 +250,40 @@ class TestForAPIFragment : Fragment(R.layout.fragment_test_for_a_p_i),
             }
 
             buttonRetrieveExposureSummary.setOnClickListener {
-                tracingVM.viewModelScope.launch {
-                    showToast(
-                        ExposureSummaryRepository.getExposureSummaryRepository()
-                            .getExposureSummaryEntities().toString()
-                    )
+                vm.launch {
+                    val summary = ExposureSummaryRepository.getExposureSummaryRepository()
+                        .getExposureSummaryEntities().toString()
+
+                    withContext(Dispatchers.Main) {
+                        showToast(summary)
+                    }
                 }
             }
 
             buttonClearDb.setOnClickListener {
-                tracingVM.viewModelScope.launch {
-                    withContext(Dispatchers.IO) {
-                        AppDatabase.getInstance(requireContext()).clearAllTables()
-                    }
+                vm.launch {
+                    AppDatabase.getInstance(requireContext()).clearAllTables()
                 }
             }
 
             buttonTracingIntervals.setOnClickListener {
-                tracingVM.viewModelScope.launch {
-                    showToast(
-                        TracingIntervalRepository.getDateRepository(requireContext()).getIntervals()
-                            .toString()
-                    )
+                vm.launch {
+                    val intervals = TracingIntervalRepository.getDateRepository(requireContext())
+                        .getIntervals()
+                        .toString()
+                    withContext(Dispatchers.Main) {
```

Comment:
Just a test fragment but maybe you want to inject the dispatcher provider anyway